StripeConnect/StripeConnect/Source/Internal/Webview/MessageSenders/SetCollectMobileFinancialConnectionsResultSender.swift

```diff
@@ -0,0 +1,110 @@
+            self.token = token
+            self.error = error
+
+            // Remove excess fields that we don't want to encode
+            self.error?._allResponseFieldsStorage = nil
+        }
+
+        // Use explicit CodingKeys instead of synthesizing so we can reference
+        // them in `keyEncodingStrategy(forKeys:)`
+        enum CodingKeys: CodingKey {
+            case financialConnectionsSession
+            case token
+            case error
+        }
+    }
+
+    /// Sends the result returned from the FinancialConnectionsSheet back to the web view
+    static func sender(value: PayloadValue) -> CallSetterWithSerializableValueSender<PayloadValue> {
+        .init(payload: .init(setter: "setCollectMobileFinancialConnectionsResult",
+                             value: value),
+              customKeyEncodingStrategy: keyEncodingStrategy)
+    }
+
+    /// Custom key encoding strategy for `PayloadValue`
+    private static func keyEncodingStrategy(forKeys keys: [any CodingKey]) -> any CodingKey {
+        /*
+         The `financialConnectionsSession` and `token` properties should encode
+         to Stripe.js types, which use snake_case.
+         Top-level payload properties should retain their camelCase encoding.
+         */
+        let lastKey = keys.last!
+
+        // Determine if the key is for a sub-property of
+        // `financialConnectionsSession` or `token`
+        guard !(lastKey is PayloadValue.CodingKeys),
+            keys.contains(where: { $0 is PayloadValue.CodingKeys }) else {
+            return lastKey
+        }
+
+        return StringCodingKey(
+            URLEncoder.convertToSnakeCase(camelCase: lastKey.stringValue)
+        )
+    }
+}
+
+extension FinancialConnectionsSheet.TokenResult {
+    /// Converts the result into one that can be sent to Stripe.js.
+    /// If an error was returned by FinancialConnections, it will be logged to analytics.
+    func toSenderValue(analyticsClient: ComponentAnalyticsClient) -> SetCollectMobileFinancialConnectionsResult.PayloadValue {
+        switch self {
+        case .completed(result: (let session, let token)):
+            return .init(
+                financialConnectionsSession: .init(accounts: session.accounts.data),
+                token: token,
+                error: nil
+            )
+
+        case .failed(error: StripeError.apiError((let apiError))):
+            // API Error
+            return .init(
+                financialConnectionsSession: nil,
+                token: nil,
+                error: apiError
+            )
+
+        case .failed(error: let error):
+            // Client error
+            analyticsClient.logClientError(error)
+            return .init(
+                financialConnectionsSession: nil,
+                token: nil,
+                error: nil
+            )
+
+        case .canceled:
+            return .init(
+                financialConnectionsSession: .init(accounts: []),
```

Comment:
I'm curious of the distinction here between initializing this with an empty list of accounts, vs a nil session (in the `.failed` case). Will these scenarios be treated differently, besides an error being logged in the latter case?

Reply:
From the stripe js documentation: https://docs.stripe.com/js/financial_connections/collect_financial_connections_accounts, https://docs.stripe.com/js/financial_connections/collect_bank_account_token

> The accounts property will contain the list of accounts collected during this session. If the user chooses to link no accounts, or exits the Authentication Flow early, the accounts array will be empty.

> If the user chooses to link no accounts, or exits the Authentication Flow early, the result object will contain: `result.financialConnectionsSession`: the updated [Financial Connections Session](https://docs.stripe.com/api/financial_connections/session) object. The accounts array will be empty.

So the calling code expects the `accounts` array to be empty when the sheet is cancelled and `null` when the sheet is errored.

Reply:
Ahh gotcha, that makes sense! Thanks for the context! 👍